vm/src/builtins/classmethod.rs

```diff
@@ -52,24 +53,38 @@ impl GetDescriptor for PyClassMethod {
     ) -> PyResult {
         let (zelf, obj) = Self::_unwrap(zelf, obj, vm)?;
         let cls = cls.unwrap_or_else(|| obj.class().clone().into());
-        Ok(PyBoundMethod::new_ref(cls, zelf.callable.clone(), &vm.ctx).into())
+        let callable = zelf.callable.lock().clone();
+        Ok(PyBoundMethod::new_ref(cls, callable, &vm.ctx).into())
     }
 }
 
 impl Constructor for PyClassMethod {
     type Args = PyObjectRef;
 
     fn py_new(cls: PyTypeRef, callable: Self::Args, vm: &VirtualMachine) -> PyResult {
-        PyClassMethod { callable }
+        PyClassMethod { callable: PyMutex::new(callable) }
             .into_ref_with_type(vm, cls)
             .map(Into::into)
     }
 }
 
+impl Initializer for PyClassMethod {
+    type Args = PyObjectRef;
+
+    fn init(
+        zelf: PyRef<Self>,
+        callable: Self::Args,
+        _vm: &VirtualMachine,
+    ) -> PyResult<()> {
+        *zelf.callable.lock() = callable;
+        Ok(())
+    }
```

Comment:
```suggestion
    fn init(
        zelf: PyRef<Self>,
        callable: Self::Args,
        _vm: &VirtualMachine,
    ) {
        *zelf.callable.lock() = callable;
    }
```

This function always returns `Ok`